### DISCUSSION TITLE
Use Nats-Expected-Last-Sequence etc. to manage competing updates more deterministically

### DISCUSSION THREAD

**Discussion by idearat** [maxpert/marmot#81] [Ideas]
In reading over some of the Nats docs I ran across a note re the two headers Nats uses to "enforce optimistic concurrency" on a stream level or a per-subject level. https://github.com/nats-io/nats-server/discussions/3772

Not sure if you're currently making use of these but perhaps they might help ensure that operations at least always replay in the proper order since those headers ensure you can't write to the stream if you're out of sequence.

Just a thought. 

Love what you're doing here!

**Comment by maxpert** [comment]
Nice! I didn't know you can do this; right now I don't but I can totally see a clear use case for publishing bunch of rows as single block in an enforced order. This will greatly reduce the inter-twining. If I can figure out mechanism of triggers to define their boundaries so that when a transaction starts Marmot can gets trigger, and when a transaction finishes marmot gets a trigger again, I can use that mechanism to with these headers to actually build global log that will preserve the order of transactions. That will let people choose the slow path for consistency mechanism. I've posted multiple times with different ideas on SQLite forums, but seems like there ain't one and there ain't gonna be one.
